### PR TITLE
feat: Server version API

### PR DESCRIPTION
**Pull request by JigarJoshi**

### CODE REVIEW

File: server/v1/api.proto [hunk 1]
```diff
@@ -578,4 +585,11 @@ service Tigris {
       body : "*"
     };
   }
+
+  rpc GetServerVersion(GetServerVersionRequest) returns
+      (GetServerVersionResponse) {
+    option (google.api.http) = {
+      get : "/api/v1/server-version"
```
Comment (efirs) [inline server/v1/api.proto:592]: I think just `version` is more intuitive and better looking:
```
/api/v1/version
```

Comment (JigarJoshi) [reply server/v1/api.proto:592]: I had that earlier. It looks like we are serving API version and not the server version in here.

Comment (efirs) [reply server/v1/api.proto:592]: This is only relevant for the manual CURLing.
And I would better remember "version",
"server-version" is something I would forget easily and would need to go to the docs.

Comment (JigarJoshi) [reply server/v1/api.proto:592]: in URL `v1` and `version` both looks really off to me.

Comment (ovaistariq) [reply server/v1/api.proto:592]: I agree that v1 is already supposed to denote to the user that it is version 1. What we  wed to show with the api is what is the full semantic version (major.minor). What should the api be named then?

Comment (efirs) [reply server/v1/api.proto:592]: v1 is API version.

just version would align with health
```
/api/v1/health
/api/v1/version
```

Should we introduce server namespace then:

```
/api/v1/server/health
/api/v1/server/version
```

Comment (JigarJoshi) [reply server/v1/api.proto:592]: yes v1 is api version and that is why to explicitly tell that we are returning server-version. alternatively we can have a generic endpoint `/api/v1/info` and serve various metadata including server-version? 

regarding server namespace - that is also good idea.

Comment (ovaistariq) [reply server/v1/api.proto:592]: See the confusion if v1 is api version then what is `/version` referring to 🙂 If we introduce the server namespace do we have to move all the endpoints under that namespace? I like the idea of an endpoint that returns metadata including version info

Comment (JigarJoshi) [reply server/v1/api.proto]: chatted offline and decided to move forward with `/info`

**Review by ovaistariq — APPROVED** [review]

File: server/v1/api.proto [hunk 2]
```diff
@@ -447,6 +447,13 @@ message StreamChange {
   bytes data = 2;
 }
 
+message GetInfoRequest {
+
+}
```
Comment (efirs) [inline server/v1/api.proto:452]: empty line in between?

Comment (JigarJoshi) [reply server/v1/api.proto:452]: I will fix it in along with next PR.

**Pull request merged**